Automattic/vip-go-mu-plugins
# Merge 'add/local-vip-go-env-const'...

## Description

When setting up a local development environment that utilizes `vip-go-mu-plugins`, it's helpful to enable debug bar and relevant tooling. Setting `WPCOM_IS_VIP_ENV` to `false` is correct for a local environment, but doesn't offer a way to selectively enable something like debug tooling. 

I went with this approach because other logic for enabling mostly revolves around things specific to the platform (employee, proxy status) that would require additional/riskier changes or inclusion of hooks/filters at the onset locally, which isn't as "clean" as defining constant for storing environment type. 

cc: @joshbetz (thanks for being a sounding board on this!)

## Changelog Description

### Feature: Enable Logged Out Debug Mode for local envs

Enables Debug mode when `WP_ENVIRONMENT_TYPE' is defined and set to `local`:

``` 
define( 'WP_ENVIRONMENT_TYPE', 'local' );
```

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
1. Check out and apply PR in local environment
2. Add to `wp-config.php`:
```
define( 'WPCOM_IS_VIP_ENV', false );
define( 'VIP_GO_ENV', 'local' );
define( 'VIP_BLOCK_WP_MAIL', true );
define( 'FILES_CLIENT_SITE_ID', 1 );
```
3. Visit local environment URL without having logged in and observe debug panel presence.
